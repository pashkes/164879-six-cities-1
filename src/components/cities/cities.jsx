import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {connect} from "react-redux";

import City from '../city-link/city-link.jsx';
import {ActionCreators} from "../../reducers/reducer";

export class Cities extends PureComponent {
  constructor(props) {
    super(props);
    this.changeCity = this.changeCity.bind(this);
  }
  changeCity(city) {
    this.props.changeCurrentCity(city);
  }
  render() {
    const {cities} = this.props;
    return (
      <section className="locations container">
        <ul className="locations__list tabs__list">
          {cities.map((city, i) => <City
            key={i}
            city={city}
            changeCity={this.changeCity}
          />)}
        </ul>
      </section>
    );
  }
}

Cities.propTypes = {
  cities: PropTypes.array.isRequired,
  changeCurrentCity: PropTypes.func.isRequired
};

const mapStateToProps = (state) => {
  return {
    cities: [...new Set([...state.offers.map((offer) => offer.city)])],
  };
};

const mapDispatchToProps = (dispatch) => ({
  changeCurrentCity: (city) => dispatch(ActionCreators.changeCity(city)),
});
export {mapStateToProps};
export default connect(mapStateToProps, mapDispatchToProps)(Cities);
