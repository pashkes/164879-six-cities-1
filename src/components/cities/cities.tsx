import React, {PureComponent} from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";

import {ActionCreators} from "../../reducer/data/data";
import City from "../city-link/city-link.jsx";

export class Cities extends PureComponent {
  constructor(props) {
    super(props);
    this._onSetCity = this._onSetCity.bind(this);
  }

  _onSetCity(city) {
    const {onChangeCity} = this.props;

    onChangeCity(city);
  }

  render() {
    const {cities} = this.props;

    return (
      <section className="locations container">
        <ul className="locations__list tabs__list">
          {cities.map((city) => <City
            key={city}
            city={city}
            onChangeCity={this._onSetCity}
          />)}
        </ul>
      </section>
    );
  }
}

Cities.propTypes = {
  cities: PropTypes.array.isRequired,
  onChangeCity: PropTypes.func.isRequired
};

const mapDispatchToProps = (dispatch) => ({
  onChangeCity: (city) => dispatch(ActionCreators.setCity(city)),
});

export default connect(null, mapDispatchToProps)(Cities);
