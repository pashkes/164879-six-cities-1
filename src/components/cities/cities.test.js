import React from 'react';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import {Cities} from './cities.jsx';

configure({adapter: new Adapter()});

const mockCities = [
  `Amsterdam`,
  `Paris`,
  `Cologne`,
  `Hamburg`,
  `Dusseldorf`
];

it(`cities renders correctly`, () => {
  const cities = shallow(<Cities cities={mockCities} onChangeCity={jest.fn} />);
  expect(cities).toMatchSnapshot();
});

