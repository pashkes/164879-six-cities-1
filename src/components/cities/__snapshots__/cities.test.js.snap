// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cities renders correctly 1`] = `
<section
  className="locations container"
>
  <ul
    className="locations__list tabs__list"
  >
    <ConnectFunction
      changeCity={[Function]}
      city="Amsterdam"
      key="0"
    />
    <ConnectFunction
      changeCity={[Function]}
      city="Paris"
      key="1"
    />
    <ConnectFunction
      changeCity={[Function]}
      city="Cologne"
      key="2"
    />
    <ConnectFunction
      changeCity={[Function]}
      city="Hamburg"
      key="3"
    />
    <ConnectFunction
      changeCity={[Function]}
      city="Dusseldorf"
      key="4"
    />
  </ul>
</section>
`;
